example/histogram.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2023 Jiri Nytra (jiri.nytra at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <flux.hpp>
+
+#include <cstddef>
+#include <ostream>
+#include <string>
+#include <random>
+#include <iostream>
+#include <iomanip>
+#include <map>
+
+template<typename T>
+flux::generator<int> randu(T min, T max)
+{
+    std::random_device rng;
+    std::uniform_int_distribution dist(min, max);
+
+    while (true)
+        co_yield dist(rng);
+}
+
+template<typename T>
+flux::generator<int> randn(T mean, T stddev = 1.0)
+{
+    std::random_device rng;
+    std::normal_distribution dist(mean, stddev);
+
+    while (true)
+        co_yield std::round(dist(rng));
+}
+
+using hist_t = std::map<int, int>;
+
+auto to_histogram = [](auto&& so_far, auto x)
+{
+    ++so_far[std::round(x)];
+    return FLUX_FWD(so_far);
```

Comment:
The forwarding in the return statement looks a bit strange to me, especially as this function returns by value.

Since we know the types involved, we can make the `so_far` parameter an rvalue reference `hist_t&&` rather than a forwarding reference, then use `std::move` rather than `FLUX_FWD` in the return statement.

(Actually I don't even know if we'd need `std::move` in the return statement -- there was a rule change about this recently, but I think it might have been in C++23?)

Reply:
I definitely agree. I changed it to rvalue reference `hist_t&&` and `std::move`, but I am also not sure if `std::move` is needed.